fossology/fossology
# feat(licenseRef): add new functionality to add new licenses and update

## Description

add functionality add new licenses and update existing licenses from SPDX.

### Changes

1. Create a new licenserefnew.json file with all the new licenses and existing license
2. Update the existing licenseref.json file with additional licenses and exceptions.

## How to test

1.  Install fossology and execute ./exportLicenseRefUsingSPDX with below options.

    -E    Update all existing licenses and also add new licenses.
          (NOTE: there may be failure of test cases)

    -e    Only update existing licenses.
          (NOTE: there may be failure of test cases)

    -n    Only add new licenses.
 
     --type Usually licenses/exceptions (optional)
          (ex: --type 'licenses')

     --url From where you want to download (optional)
         (ex: --url 'https://spdx.org/licenses/licenses.json')

check if each option works properly. Once a new licenserefnew.json file gets created. copy this file to ilstall/db and run fo-postinstall scripts. check if the scripts run successfully.
